### PR TITLE
Assignments api

### PR DESCRIPTION
**Pull request by gadarajainik**

### CODE REVIEW

File: academiverse-api/src/main/java/com/academiverse/academiverse_api/dto/request/AssignmentForStudentGetRequest.java
Comment (gadarajainik) [inline academiverse-api/src/main/java/com/academiverse/academiverse_api/dto/request/AssignmentForStudentGetRequest.java:1]: Is this file even used?

File: academiverse-api/src/main/java/com/academiverse/academiverse_api/controller/AssignmentController.java [hunk 1]
```diff
@@ -55,4 +55,15 @@ public ResponseEntity<BaseResponse> getAssignmentsForInstruct(@PathVariable Long
     public ResponseEntity<BaseResponse> activateAssignment(@PathVariable Long assignmentId) {
         return ResponseEntity.ok().body(assignmentService.activateAssignment(assignmentId));
     }
+
+    @PostMapping("/submit")
+    public ResponseEntity<BaseResponse> saveAssignmentSubmission(@RequestBody AssignmentSubmitRequest assignmentSubmitRequest) {
+        return ResponseEntity.ok().body(assignmentService.submitAssignment(assignmentSubmitRequest));
+    }
+
+    @GetMapping("/submittedAssignment")
```
Comment (gadarajainik) [inline academiverse-api/src/main/java/com/academiverse/academiverse_api/controller/AssignmentController.java]: What is the need of this API endpoint?

File: academiverse-ui/src/app/(routes)/courses/assignments/page.js [hunk 2]
```diff
@@ -47,20 +47,43 @@ const AssignmentPage = () => {
...
-    if (!res.isError) {
-      const formattedData = res.data.map((assignment) => ({
-        id: assignment.assignmentId,
-        title: assignment.assignmentTitle,
-        description: assignment.assignmentDescription,
-        dueDate: assignment.assignmentDueDate,
-        totalMarks: assignment.totalMarks
-      }));
-      setAssignments(formattedData);
+    if(session?.userDetails?.role === 'student'){
```
Comment (gadarajainik) [inline academiverse-ui/src/app/(routes)/courses/assignments/page.js]: Why IF-ELSE based on user role?

Comment (hetvijoshi) [reply academiverse-ui/src/app/(routes)/courses/assignments/page.js]: because the assignments page is different for both the user roles

**Pull request merged**